src/Sharpliner/AzureDevOps/AzureDevOpsDefinition.cs

```diff
@@ -882,7 +882,7 @@ protected static Parameter StageListParameter(string name, string? displayName =
     /// AzDO allows an empty dependsOn which then forces the stage/job to kick off in parallel.
     /// If dependsOn is omitted, stages/jobs run in the order they are defined.
     /// </summary>
-    protected static DependsOn NoDependsOn => [];
+    protected static DependsOn NoDependsOn => [ string.Empty ];
```

Comment:
What didn't work there? It would be better, if supplying `[]` worked because I can imagine people might not even find the `NoDependsOn` and will just put `[]`

Reply:
I think we ignore empty collections

Reply:
Can't we tell the properties through the `YamlMember` attribute to not serialize `null` only and make these `null` by default? So that `[]`serializes as `[]`, Count == 1 serializes as string via the `IYamlConvertible` and no value makes the property not appear?

Reply:
https://github.com/sharpliner/sharpliner/blob/58c12072b8f24471ebfc66aa81e41b17c95359a2/src/Sharpliner/SharplinerSerializer.cs#L23

we globally ignore empty collections, I think that changing this to include empty collections will change too much

Reply:
I don't follow - my proposal is to go to stage and job where there are DependsOn members, making them nullable and changing their attribute by setting OmitWhenNull.
Won't that work?

Reply:
I got this to work, 

@premun I just looked at the docs, I wasn't able to find reference to having a `dependsOn: ''` I only found references showing having `dependsOn: []` to allow parallel